### PR TITLE
Add `GetRemoteCandidates()`

### PR DESCRIPTION
**Pull request by stv0g**
This is analogue to `GetLocalCandidates()` and provides the user with the ability to get a slice of previously added remote candidates without keeping track of them in the application code.

### CODE REVIEW

**Review by at-wat — COMMENTED** [review]
The code looks good!
It would be better to add a test if it's not hard.

**Comment by stv0g** [comment]
Hi @at-wat,

sure. I also added one for `GetLocalCandidates()`.

File: agent_test.go [hunk 1]
```diff
@@ -1322,6 +1322,74 @@ func TestGetRemoteCredentials(t *testing.T) {
...
+			Port:      1000 + i,
+			Component: 1,
+		}
+
+		cand, errCand := NewCandidateHost(&cfg)
+		assert.NoError(t, errCand)
+
+		expectedCandidates = append(expectedCandidates, cand)
+
+		err = a.addCandidate(context.Background(), cand, dummyConn)
```
Comment (at-wat) [inline agent_test.go:1382]: Lint error

Comment (stv0g) [reply agent_test.go:1382]: Thanks :) Its fixed in the latest version

**Review by at-wat — APPROVED** [review]
LGTM

**Pull request merged**